gnosis/eth/eip712/__init__.py

```diff
@@ -75,15 +76,16 @@ def _encode_field(name, typ, value):
             return ["bytes32", fast_keccak(value)]
 
         if typ.endswith("]"):
-            parsed_type = typ[:-2]
-            type_value_pairs = dict(
-                [_encode_field(name, parsed_type, v) for v in value]
-            )
-            h = fast_keccak(
-                encode_abi(
-                    list(type_value_pairs.keys()), list(type_value_pairs.values())
-                )
-            )
+            # Array type
+            parsed_type = typ[: typ.rindex("[")]
+            type_value_pairs = [_encode_field(name, parsed_type, v) for v in value]
+            # Empty array
+            if value:
```

Comment:
I suggest to check if value exists before to get the `type_value_pairs` you can do something like:
```
     if value:
          parsed_type = typ[: typ.rindex("[")]
          type_value_pairs = [_encode_field(name, parsed_type, v) for v in value]
          data_types, data_hashes = zip(*type_value_pairs)
      else: 
          data_types, data_hashes = [], []
```

Reply:
Good suggestion, more optimal!

Reply:
Done